java/com/google/flatbuffers/Table.java

```diff
@@ -273,6 +284,228 @@ protected static int compareStrings(int offset_1, byte[] key, ByteBuffer bb) {
+    @Override
+    public int getValue(ByteBuffer bb, int offsetInByteBuffer) {
+      return bb.get(offsetInByteBuffer);
+    }
+  };
+
+  static FieldAsIntGetter UBYTE_FIELD_GETTER = new FieldAsIntGetter() {
+    @Override
+    public int getValue(ByteBuffer bb, int offsetInByteBuffer) {
+      return Unsigneds.asComparable( bb.get(offsetInByteBuffer) );
+    }
+  };
+
+  static FieldAsIntGetter SHORT_FIELD_GETTER = new FieldAsIntGetter() {
+    @Override
+    public int getValue(ByteBuffer bb, int offsetInByteBuffer) {
+      return bb.getShort(offsetInByteBuffer);
+    }
+  };
+
+  static FieldAsIntGetter USHORT_FIELD_GETTER = new FieldAsIntGetter() {
+    @Override
+    public int getValue(ByteBuffer bb, int offsetInByteBuffer) {
+      return Unsigneds.asComparable( bb.getShort(offsetInByteBuffer) );
+    }
+  };
+
+  static FieldAsIntGetter INT_FIELD_GETTER = new FieldAsIntGetter() {
+    @Override
+    public int getValue(ByteBuffer bb, int offsetInByteBuffer) {
+      return bb.getInt(offsetInByteBuffer);
+    }
+  };
+
+  static FieldAsIntGetter UINT_FIELD_GETTER = new FieldAsIntGetter() {
+    @Override
+    public int getValue(ByteBuffer bb, int offsetInByteBuffer) {
+      return Unsigneds.asComparable( bb.getInt(offsetInByteBuffer) );
+    }
+  };
+
+  static FieldAsLongGetter LONG_FIELD_GETTER = new FieldAsLongGetter() {
+    @Override
+    public long getValue(ByteBuffer bb, int offsetInByteBuffer) {
+      return bb.getLong(offsetInByteBuffer);
+    }
+  };
+
+  static FieldAsLongGetter ULONG_FIELD_GETTER = new FieldAsLongGetter() {
+    @Override
+    public long getValue(ByteBuffer bb, int offsetInByteBuffer) {
+      return Unsigneds.asComparable( bb.getLong(offsetInByteBuffer) );
+    }
+  };
+
+  static FieldAsIntGetter FLOAT_FIELD_GETTER = new FieldAsIntGetter() {
+    @Override
+    public int getValue(ByteBuffer bb, int offsetInByteBuffer) {
+      return Floats.asComparable( bb.getFloat(offsetInByteBuffer) );
+    }
+  };
+
+  static FieldAsLongGetter DOUBLE_FIELD_GETTER = new FieldAsLongGetter() {
+    @Override
+    public long getValue(ByteBuffer bb, int offsetInByteBuffer) {
+      return Floats.asComparable( bb.getDouble(offsetInByteBuffer) );
+    }
+  };
+
+
+  private int __genericIntLookupByKey(int vectorFieldOffset, int keyFieldOffset, int key, int defaultValue,
+                                      FieldAsIntGetter fieldGetter ) {
+    int fieldDataOffset = __offset(vectorFieldOffset);
+    if ( fieldDataOffset == 0 )
+      return 0;
+    int vectorLocation = __vector( fieldDataOffset );
+    int span = bb.getInt(vectorLocation-SIZEOF_INT);
+    int start = 0;
+    while (span != 0) {
+      int middle = span / 2;
+      int tableOffset = __indirect(vectorLocation + SIZEOF_INT * (start + middle), bb);
+      int keyValueOffset = __offset( keyFieldOffset, tableOffset, bb );
+      int val = keyValueOffset != 0 ? fieldGetter.getValue(bb, tableOffset + keyValueOffset) : defaultValue;
```

Comment:
Going via this `fieldGetter` is slower, and requires us to convert floats to ints etc.. all to avoid generating code for this function? The generated code version was fast, simple, and no duplicated code to maintain.